### PR TITLE
chore(sliding sync): add logs for the limited flag computation

### PR DESCRIPTION
**Pull request by bnjbvr**
This should help us debug issues with false negatives.

### CODE REVIEW

File: crates/matrix-sdk/src/sliding_sync/mod.rs [hunk 1]
```diff
@@ -847,6 +847,7 @@ impl StickyData for SlidingSyncStickyParameters {
 /// TODO remove this workaround as soon as support of the `limited` flag is
 /// properly implemented in the open-source proxy: https://github.com/matrix-org/sliding-sync/issues/197
 // NOTE: SS proxy workaround.
+#[instrument(level = "trace", skip_all)]
```
Comment (jplatte) [inline crates/matrix-sdk/src/sliding_sync/mod.rs:850]: The span isn't usually printed when no event happens within, so I don't think overriding the default level here is really necessary. That's not to say it's bad though.

File: crates/matrix-sdk/src/sliding_sync/mod.rs [hunk 2]
```diff
@@ -859,10 +860,12 @@ fn compute_limited(
         }
 
         if room.limited {
-            // If the room was already marked as limited, the server knew more than we did.
+            // If the room was already marked as limited, the server knew more than we do.
             continue;
         }
 
+        trace!(room_id = ?room_id, "starting for room");
```
Comment (jplatte) [inline crates/matrix-sdk/src/sliding_sync/mod.rs:867]:
```suggestion
        trace!(?room_id, "starting");
```

**Review by jplatte — APPROVED** [review]

File: crates/matrix-sdk-ui/tests/integration/timeline/queue.rs [hunk 3]
```diff
@@ -228,8 +228,12 @@ async fn clear_with_echoes() {
     // Send a message without mocking the server response.
     timeline.send(RoomMessageEventContent::text_plain("Send failure").into(), None).await;
 
-    // Wait for the first message to fail.
-    sleep(Duration::from_millis(10)).await;
+    // Wait for the first message to fail. Don't use time, but listen for the first
+    // timeline item diff to get back signalling the error.
+    {
+        let (_, mut timeline_stream) = timeline.subscribe().await;
```
Comment (jplatte) [inline crates/matrix-sdk-ui/tests/integration/timeline/queue.rs:234]: This is racy, the stream should be created before the call to `timeline.send` above.

Comment (jplatte) [reply crates/matrix-sdk-ui/tests/integration/timeline/queue.rs:234]: Alternatively, the first tuple element could be not ignored here and checked for the `SendingFailed` local echo state, but that will be more complicated.

Comment (bnjbvr) [reply crates/matrix-sdk-ui/tests/integration/timeline/queue.rs:230]: Is the new version more robust?

Comment (jplatte) [reply crates/matrix-sdk-ui/tests/integration/timeline/queue.rs:230]: Yes, that's what I was thinking of! 👍🏼

**Pull request merged**